src/storm/storage/prism/Update.cpp

```diff
@@ -79,10 +83,9 @@ Update Update::substitute(std::map<storm::expressions::Variable, storm::expressi
     for (auto const& assignment : this->getAssignments()) {
         newAssignments.emplace_back(assignment.substitute(substitution));
     }
-    auto newLikelihoodExpression = this->getLikelihoodExpression().substitute(substitution);
-    STORM_LOG_THROW(likelihoodExpression.containsVariables() || likelihoodExpression.evaluateAsRational() >= 0, storm::exceptions::IllegalArgumentException,
+    auto newLikelihoodExpression = this->getLikelihoodExpression().substitute(substitution).simplify();
+    STORM_LOG_THROW(!newLikelihoodExpression.isLiteral() || newLikelihoodExpression.evaluateAsRational() >= 0, storm::exceptions::IllegalArgumentException,
                     "Substitution yielding negative probabilities in '" << this->getLikelihoodExpression() << "' are not allowed.");
-    // FIXME: The expression could be simplified, but 1/K (where K is an int) is then resolved to 0, which is incorrect (for probabilities).
```

Comment:
Note: This old comment no longer applies since the `simplify()` function no longer simplifies divisions with a remainder